Class8_Sangavi

```diff
@@ -0,0 +1,71 @@
+-----------------------Number Check-------------------------------------
+import string
+import re
+
+def numcheck(str1):
+    if str == "":
+        return False
+    elif str1.count(".")>1 or str1.endswith("."):
+        return False
+    elif str1.count("-") and str1.count("+") == 1:
+        return False
+    elif str1.count("-") or str1.count("+") == 1:
+        if str1.startswith("-") or str1.startswith("+"):
+            return True
+    else:
+        return False
+    return True
+
+str1 = "+123001.29"
+print numcheck(str1)
+
+
+s = re.match(r'[+|-]?[0-9]+[\.]?[0-9]+$',str1)     #regex pattern
```

Comment:
Can you write this program without regex?? or can you explain what the regex.match will do? say if you have write the match() function, how it will be?